store/tikv/coprocessor.go

```diff
@@ -621,7 +621,8 @@ func (it *copIterator) handleTaskOnce(bo *Backoffer, task *copTask, ch chan copR
 }
 
 func (it *copIterator) handleCopStreamResult(bo *Backoffer, stream *tikvrpc.CopStreamResponse, task *copTask, ch chan copResponse) ([]*copTask, error) {
-	var resp *coprocessor.Response
+	defer stream.Close()
+	var resp, lastResp *coprocessor.Response
```

Comment:
We can remove `lastResp` and use a temporary variable in line 632.

Reply:
1. one variable is not enough to record both last response and the response of current one.
2. there is a issue @hicqu told me yesterday, that TiKV don't set the range field when error field is not null (mocktikv always set range field, so I have to adjust here, anyway)

I'd like to modify here and also mocktikv's behavior after this PR.